tests/unit/extensions/test_extensions.py

```diff
@@ -134,3 +134,102 @@ def test_project_load_extensions(fake_extensions, tmp_path):
 
     # New part
     assert parts[f"{FullExtension.NAME}/new-part"] == {"plugin": "nil", "source": None}
+
+
+def test_flask_extensions(fake_extensions, tmp_path, input_yaml, monkeypatch):
```

Comment:
Thanks for the tests! I think a couple are missing:
- A test that checks that `apply_extensions()` fails if `requirements.txt` is missing
- A test that checks that `build_base: ubuntu:22.04` gets added if the key is missing and `base` is bare
- Since the extension has a custom merge for the parts, a test that merges existing keys. For example: a yaml that has:
```
parts:
    flask/dependencies:
        python-requirements: ["requirements-jammy.txt"]

Reply:
In light of this, I think all of these tests (and these new ones) should be moved to a `test_flask` module in `unit/extensions`

Reply:
Gotcha, I have included more tests and move they to the `test_flask.py`, thanks!